OneCommunityGlobal/HighestGoodNetworkApp
# yihan add team code permission

# Description
<img width="721" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94303659/f0484dd7-856c-4dba-911f-994b9cee100b">

## Related PRS (if any):
The related backend PR is [#494](https://github.com/OneCommunityGlobal/HGNRest/pull/494)
This is the 2nd PR of #1162 which hasn't been merged to dev yet.

## Main changes explained:
- Update file RolePermissions.jsx and UserRoleTab.jsx for adding Edit Team 4-Digit Codes permission to permission management page.
- Update file UserProfile.jsx and WeeklySummariesReport.jsx for allowing edit team code section if the user is owner or has that permission.
- Update file FormattedReport.jsx and create TeamCode.jsx for separating TeamCode component from FormattedReport.jsx

## How to test:
1. check into current branch
2. do `npm run start:local` to run this PR locally
3. log as Owner user
a. go to any user's profile page -> Teams Tab -> Team Code section, or Reports -> Weekly Summaries Report
b. verify if you can edit Team Code input field.
c. go to Other Links -> Permissions Management -> Manage User Permissions to add Edit Team 4-Digit Codes to a user who is not Owner.
4. log as that not-owner user who has Edit Team 4-Digit Codes permission
5. repeat step 3.a to see if this user can edit team code section.
6. log as any not-owner user who doesn't haveEdit Team 4-Digit Codespermission
7. repeat step 3.a, verify if the user cannot edit team code section.

## Screenshots or videos of changes:

1. Owner and other users who has Edit Team 4-Digit Codes permission:
<img width="1226" alt="截屏2023-08-31 15 28 01" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94303659/c99725a2-2899-4f09-8f57-53dfc3e6041d">


3. Other users who doesn't have Edit Team 4-Digit Codes permission:
<img width="1222" alt="截屏2023-08-31 15 24 03" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94303659/52a671f6-bf95-4704-9f44-e1c4858e3803">
